From 7f40c5995d4e4b66edbf85ae491eb427a14cedb4 Mon Sep 17 00:00:00 2001
From: Philippe Sauter <phsauter@iis.ee.ethz.ch>
Date: Tue, 20 Aug 2024 11:25:13 +0200
Subject: [PATCH] multdiv tieoff for RV32MNone case

---
 ibex_ex_block.sv | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ibex_ex_block.sv b/ibex_ex_block.sv
index 62e03964..65af9e6c 100644
--- a/ibex_ex_block.sv
+++ b/ibex_ex_block.sv
@@ -189,6 +189,11 @@ module ibex_ex_block #(
         .valid_o            ( multdiv_valid         ),
         .multdiv_result_o   ( multdiv_result        )
     );
+  end else if (RV32M == RV32MNone) begin : gen_multidiv_none
+    assign multdiv_alu_operand_a = '0;
+    assign multdiv_alu_operand_b = '0;
+    assign multdiv_result = '0;
+    assign multdiv_valid = 1'b0;
   end
 
   // Multiplier/divider may require multiple cycles. The ALU output is valid in the same cycle
@@ -196,4 +201,4 @@ module ibex_ex_block #(
   // final cycle of ALU operation).
   assign ex_valid_o = multdiv_sel ? multdiv_valid : ~(|alu_imd_val_we);
 
-endmodule
+endmodule
\ No newline at end of file
-- 
2.25.1

