From c4eb2f59e5f33fb62559a8300b8c5f931722d825 Mon Sep 17 00:00:00 2001
From: Philippe Sauter <phsauter@iis.ee.ethz.ch>
Date: Fri, 19 Dec 2025 17:02:58 +0100
Subject: [PATCH] add multdiv unused signal tieoff

---
 cve2_ex_block.sv | 9 ++++++++
 1 file changed, 9 insertions(+)

diff --git a/cve2_ex_block.sv b/cve2_ex_block.sv
index f84d990f..45a8e35c 100644
--- a/cve2_ex_block.sv
+++ b/cve2_ex_block.sv
@@ -166,6 +166,15 @@ module cve2_ex_block #(
       .valid_o           (multdiv_valid),
       .multdiv_result_o  (multdiv_result)
     );
+  end else begin : gen_multdiv_tieoff
+    // tie off signals when multiplier/divider is not present
+    assign multdiv_alu_operand_a = '0;
+    assign multdiv_alu_operand_b = '0;
+    assign multdiv_valid         = '0;
+    assign multdiv_result        = '0;
+    assign multdiv_imd_val_d[0]  = '0;
+    assign multdiv_imd_val_d[1]  = '0;
+    assign multdiv_imd_val_we    = '0;
   end
 
   // Multiplier/divider may require multiple cycles. The ALU output is valid in the same cycle
-- 
2.43.5

